nncf/model_creation.py

```diff
@@ -136,17 +103,18 @@ def create_compressed_model(model: Module, config: NNCFConfig,
                                    scopes_without_shape_matching=scopes_without_shape_matching)
 
     should_init = resuming_state_dict is None
-    compression_algo_builder_list = create_compression_algorithm_builders(config, should_init=should_init)
+    composite_builder = CompositeCompressionAlgorithmBuilder(config, should_init=should_init)
+
+    composite_builder.apply_to(compressed_model)
+    composite_builder.handle_frozen_layers()
```

Comment:
Applying the algorithm without handling frozen layers should not be allowed. The `handle_frozen_layers` call should be moved inside `apply_to`, and the method itself be made protected.

Reply:
done